src/verify.c

```diff
@@ -770,6 +770,51 @@ static enum swupd_code deal_with_extra_files(struct manifest *manifest, bool fix
 	return ret;
 }
 
+static struct list *sanitize_files(struct list *bundles_files, struct list *all_files)
```

Comment:
That's another function that is a candidate to be a generic list function.

    // return a list with the elements of listA that are present in listB
    // listA and listB are unchanged
    struct list *filter_list(struct list *listA, struct list *listB, comp_fn_t comp_fn);

So you can use it like this:

    listC = filter_list(listA, listB, cmp_fn);
    free_list_and_data(listB, free_fn);
    // and just use listC
    free_list_and_data(listA, free_fn);
    free_list(listC, free_fn);

Or you can implement it in a way that it changes the listA and returns it. But in this case you need to free elements and data that aren't used:

    void filter_list(struct list *listA, struct list *listB, comp_fn_t comp_fn, list_free_data_fn_t free_fn);
    filter_list(listA, listB, cmp_fn, free_fn);
    free_list_and_data(listB, free_fn);
    // and just use listA
    free_list_and_data(listA, free_fn);

I think I prefer the second (I suggested that in other comments too) but the first one is good too.

Reply:
I think the purpose of this function is being misunderstood. It's purpose is not to get the elements of listA that are present in listB, its purpose is only to make sure the files in listA marked as deleted, are not needed by another installed bundle. I don't see it being useful as a generic function.

Reply:
this function does a loop in the list, applies a condition and if condition is true we add that element to the returned list. The condition in this case is what you described. And this code is repeated several times in swupd with different implementations. I suggested having a generic list function (similar to python filter https://docs.python.org/3/library/functions.html#filter) to execute that.

Reply:
Sorry, I pointed the wrong function. Not sure if python lists have one implementation on what I thought, but I know we could use that in filter_out_existing_files() too

Reply:
I think I found a way to make it generic, I'll implement that